### PR TITLE
Add greedy time-constrained ORC-WER

### PR DESCRIPTION
**Pull request by thequilo**

### CODE REVIEW

**Comment by boeddeker** [comment]
We should add again a print in the cli. e.g. printing the files that are written and/or a summary of the WER.

File: meeteval/wer/matching/cy_greedy_combination_matching.pyx [hunk 1]
```diff
@@ -41,3 +41,51 @@ def cy_forward_col(
...
+    npy_float64[:] b_begin not None,
+    npy_float64[:] b_end not None,
+    uint cost_substitution = 1,
+) -> np.ndarray:
+    """
+    Args:
+        column: The column to be updated
+        a: Sequence in column direction (make sure that `len(column) == len(a) + 1`! otherwise SEGFAULT!!)
+        b: Sequence in row direction. This function updates `column` `len(b)` times
+        tmp: Pre-allocated temporary memory. Must have the same shape as `column`
```
Comment (boeddeker) [inline meeteval/wer/matching/cy_greedy_combination_matching.pyx:63]: Did you change the code and removed tmp from the signature?

File: meeteval/wer/matching/cy_greedy_combination_matching.pyx [hunk 2]
```diff
@@ -41,3 +41,51 @@ def cy_forward_col(
...
+    tmp[0, ...] = column
+    current = 0
+    for j in range(b.shape[0]):
+        current = (j + 1) % 2
+        prev = j % 2
+        b_ = b[j]
+        b_begin_ = b_begin[j]
+        b_end_ = b_end[j]
+
+        # TODO: can we swap without copy?
```
Comment (boeddeker) [inline meeteval/wer/matching/cy_greedy_combination_matching.pyx:78]: I don't get the meaning of that statement. Where is a swap operation?

File: meeteval/wer/matching/cy_greedy_combination_matching.pyx (hunk 1 above)
Comment (thequilo) [reply meeteval/wer/matching/cy_greedy_combination_matching.pyx:63]: Yes, I forgot to remove that. The preallocation didn't improve the runtime much

Comment (thequilo) [reply meeteval/wer/matching/cy_greedy_combination_matching.pyx:63]: Removed

File: meeteval/wer/matching/cy_greedy_combination_matching.pyx (hunk 2 above)
Comment (thequilo) [reply meeteval/wer/matching/cy_greedy_combination_matching.pyx:78]: That was again an old comment... Removed

File: meeteval/wer/matching/cy_greedy_combination_matching.pyx [hunk 3]
```diff
@@ -41,3 +41,51 @@ def cy_forward_col(
...
+        tmp[current, 0] = tmp[prev, 0] + 1
+        for i in range(1, a.shape[0] + 1):
+            a_ = a[i - 1]
+            if a_begin[i - 1] >= b_end_ or b_begin_ >= a_end[i - 1]:
```
[the remark this answers is not in the document]
Comment (thequilo) [reply meeteval/wer/matching/cy_greedy_combination_matching.pyx:82]: Yes, we decided for this in the time-constrained Levenshtein distance inn `levenshtein.h` and now everything is tested against that. I think the argument was that two time points should not overlap

**Comment by thequilo** [comment]
> We should add again a print in the cli. e.g. printing the files that are written and/or a summary of the WER.

Yes, I agree. But that should be a separate PR.

File: meeteval/wer/matching/cy_greedy_combination_matching.pyx (hunk 3 above)
[the remark this answers is not in the document]
Comment (boeddeker) [reply meeteval/wer/matching/cy_greedy_combination_matching.pyx:78]: ok. I just wanted to ensure, that it is the same.

**Review by boeddeker — APPROVED** [review]

**Pull request merged**